clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
I think the semantics of subscribe(empty-list) should be similar to pause(all), but leave the consumer still registered as member of the group with coordinator; for unsubscribe() the consumer means "do not talk to coordinator anymore" and moving forward we may add a leave-group request (there is already a ticket I think).

As for now let's keep the original approach to check the assignment upon each commit call; thoughts? @hachikuji @onurkaraman